code/_onclick/hud/new_player.dm

```diff
@@ -11,19 +11,21 @@
 
 	var/list/buttons = subtypesof(/atom/movable/screen/lobby)
 	for(var/button_type in buttons)
-		var/atom/movable/screen/lobby/lobbyscreen = new button_type()
+		var/atom/movable/screen/lobby/lobbyscreen = new button_type(null, src, owner)
 		lobbyscreen.SlowInit()
-		lobbyscreen.hud = src
 		static_inventory += lobbyscreen
-		if(istype(lobbyscreen, /atom/movable/screen/lobby/button))
-			var/atom/movable/screen/lobby/button/lobby_button = lobbyscreen
-			lobby_button.owner = REF(owner)
 
 /atom/movable/screen/lobby
 	plane = SPLASHSCREEN_PLANE
 	layer = LOBBY_BUTTON_LAYER
 	screen_loc = "TOP,CENTER"
 
+INITIALIZE_IMMEDIATE(/atom/movable/screen/lobby)
+
+/atom/movable/screen/lobby/Initialize(mapload, datum/hud/new_player/_hud)
+	. = ..()
+	hud = _hud
```

Comment:
This should really be done like this:

```suggestion
/atom/movable/screen/lobby/Initialize(mapload, datum/hud/new_player/hud)
	. = ..()
	src.hud = hud
```

per https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar